### PR TITLE
Add validation for disabled services

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1050

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 4
           

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ServiceDisabledException.java  1
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForDisabledService.java  1
         

Clones added
============
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForDisabledService.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5044694&bid=16612538)

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 4
           

Complexity increasing per file
==============================
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForDisabledService.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ServiceDisabledException.java  1
         

Clones removed
==============
+ src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForFailedStatus.java  -1
+ src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForDisabledPayments.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5044694&bid=16612538)

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 4
           

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ServiceDisabledException.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java  1
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForDisabledService.java  1
         

Clones removed
==============
+ src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForDisabledPayments.java  -1
+ src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForFailedStatus.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5044694&bid=16612538)

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 4
           

Complexity increasing per file
==============================
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForDisabledService.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ServiceDisabledException.java  1
         

Clones removed
==============
+ src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForDisabledPayments.java  -1
+ src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForFailedStatus.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5044694&bid=16612538)

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/common/Event.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
     ZIPFILE_PROCESSING_STARTED, // when processor starts processing zipfile from blob
     DOC_FAILURE, // generic failure while processing zip file. before uploading to document management
     FILE_VALIDATION_FAILURE,
+    SERVICE_DISABLED,
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/common/Event.java]: SERVICE_DISABLED is name for a property. Name for an event should be DISABLED_SERVICE_FAILURE or something like this.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/out/msg/ErrorCode.java [hunk 2]
```diff
@@ -5,6 +5,7 @@
     ERR_FILE_LIMIT_EXCEEDED, // size too big
     ERR_METAFILE_INVALID,
     ERR_PAYMENTS_DISABLED, //payments not allowed for the container or not allowed in specific environment
+    ERR_SERVICE_DISABLED, // service is disabled in specific environment
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/out/msg/ErrorCode.java]: And this name os ok

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java [hunk 3]
```diff
@@ -322,6 +324,12 @@ private ZipFileProcessingResult processZipFileContent(
             );
             handleInvalidFileError(Event.FILE_VALIDATION_FAILURE, containerName, zipFilename, leaseId, ex);
             return null;
+        } catch (ServiceDisabledException ex) {
+            log.error(
+                "Rejected file {} from container {} - Service is disabled", zipFilename, containerName
+            );
+            handleInvalidFileError(Event.SERVICE_DISABLED, containerName, zipFilename, leaseId, ex);
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java]: Makes sense to rename this method to 'handleFileUploadError' or something like this - not 100% sure

File: src/main/resources/application.yaml [hunk 4]
```diff
@@ -121,32 +121,38 @@ containers:
     - container: sscs
       jurisdiction: SSCS
       poBox: 12626
+      enabled: true
```
Comment (apereverzin) [inline src/main/resources/application.yaml]: Is it possible to have true as the default value? As far as I remember we discussed this

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/common/Event.java (hunk 1 above)
Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/common/Event.java]: Renamed

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java (hunk 3 above)
Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java]: This method is used for all zip file failure types. I prefer to leave as it is now

File: src/main/resources/application.yaml (hunk 4 above)
Comment (aliveni) [reply src/main/resources/application.yaml]: Updated code to set the values as true by default

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 4
           

Complexity increasing per file
==============================
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForDisabledService.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ServiceDisabledException.java  1
         

Clones removed
==============
+ src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForDisabledPayments.java  -1
+ src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForFailedStatus.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5044694&bid=16612538)

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 4
           

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java  1
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForDisabledService.java  1
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ServiceDisabledException.java  1
         

Clones removed
==============
+ src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForFailedStatus.java  -1
+ src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForDisabledPayments.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5044694&bid=16612538)

**Pull request merged**